### PR TITLE
Run migrations within the Kubermatic controller manager

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Migrations are now executed only after the leader lease was acquired
```

### CODE REVIEW

File: api/pkg/crd/migrations/migrations.go [hunk 1]
```diff
@@ -35,51 +31,51 @@ type cleanupContext struct {
...
 	if err != nil {
-		glog.Fatal(err)
+		return fmt.Errorf("failed to create kubeClient: %v", err)
+	}
+	kubermatiClient, err := kubermaticclientset.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to create Kuermatic client: %v", err)
 	}
 
-	ctx.config.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: scheme.Codecs}
```
Comment (alvaroaleman) [inline api/pkg/crd/migrations/migrations.go:65]: I removed this, do you think this will cause any issues? I think this is the default anyways

Comment (mrIncompetent) [reply api/pkg/crd/migrations/migrations.go:65]: I don't think so

**Pull request merged**